frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/RediscoverSuggester.java

```diff
@@ -62,13 +71,49 @@ private void queueSuggestion(Path macheteFilePath) {
       queueDiscoverOperation.run();
     } else { // closing dialog goes here too
       LOG.info("Rediscover declined from dialog");
-      macheteFilePath.setFileModificationDate(System.currentTimeMillis());
+      setMacheteFileModificationDate(macheteFilePath, System.currentTimeMillis());
     }
   }
 
+  @UIEffect
+  private void setMacheteFileModificationDate(Path macheteFilePath, long timeMillis) {
+    macheteFilePath.setFileModificationDate(timeMillis);
+  }
+
   private long daysDiffTillNow(long lastModifiedTimeMillis) {
     val currentTimeMillis = System.currentTimeMillis();
     val millisDiff = currentTimeMillis - lastModifiedTimeMillis;
     return millisDiff / (24 * 60 * 60 * 1000);
   }
+
+  public void enqueueChecksAndSuggestIfApplicable(Path macheteFilePath) {
+    new Task.Backgroundable(
+        gitRepository.getProject(),
+        getString("string.GitMachete.RediscoverSuggester.backgroundable-check-task.title")) {
+      @Override
+      public void run(ProgressIndicator indicator) {
+        if (!areAllLocalBranchesManaged(macheteFilePath)) {
+          ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> queueSuggestion(macheteFilePath));
+        } else {
+          ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL,
+              () -> setMacheteFileModificationDate(macheteFilePath, System.currentTimeMillis()));
```

Comment:
Oh, so you basically defer the next check until `DAYS_AFTER_WHICH_TO_SUGGEST_DISCOVER` pass right? Pls add a comment in code